lib/workers/repository/package-files.ts

```diff
@@ -71,6 +138,52 @@ export class PackageFiles {
       deps += pad ? '</blockquote>\n</details>\n\n' : '';
     }
 
-    return title + deps;
+    return (setTitle ? title : '') + deps;
+  }
+
+  private static backup(): () => void {
+    const backup = this.data; // backup data
+    // deep clone data
+    this.data = new Map(clone(Array.from(this.data))); // only mutate cloned data
+
+    return () => {
+      this.data = backup;
+    };
+  }
+
+  /**
+   * Removes the last dependency/entry in the PackageFiles data
+   * i.e. the last line in the tobe generated detected dependency section
+   * @Returns true if anything that translates to a markdown written line was deleted
+   *          otherwise false is returned
+   */
+  private static pop(): boolean {
+    // get detected managers list of the last listed base branch
+    const [branch, managers] = Array.from(this.data).pop() ?? [];
+    if (!branch) {
+      return false;
+    }
+
+    // delete base branch listing if it has no managers left
+    if (!managers || is.emptyObject(managers)) {
+      return this.data.delete(branch);
+    }
+
+    // get all manifest files for the last listed manager
+    const [manager, packageFiles] = Object.entries(managers).pop() ?? [];
+
+    // delete current manager if it has no manifest files left
+    if (!packageFiles || is.emptyArray(packageFiles)) {
+      return delete managers[manager!];
+    }
+
+    // delete manifest file if it has no deps left
+    const len = packageFiles.length - 1;
+    if (is.emptyArray(packageFiles[len].deps)) {
+      return !!packageFiles.pop();
+    }
+
+    // remove the last listed dependency
+    return !!packageFiles[len].deps.pop();
```

Comment:
Not something I expect you to fix now, but is there some bin packing optimization you can do here?

Reply:
current solution is O(n^2) where n = num of dependencies, if performance is an issue I had another solution (albeit less elegant) with linear time complexity (hidden constant = 1) where we straight truncate the generated markdown and then fixing its missing html closing tags using bracket balancing algorithm.

I think the added performance is negligible in our context where we make a lot of fs reads/writes and api calls. whats more is that usually markdown truncation is unnecessary and when it is, we probably dealing with a big monorepo which makes this optimization even more negligible